test/boost/memtable_test.cc

```diff
@@ -984,9 +984,9 @@ SEASTAR_TEST_CASE(failed_flush_prevents_writes) {
 
         auto failed_memtables_flushes_count = db.cf_stats()->failed_memtables_flushes_count;
 
-        utils::get_local_injector().enable("table_seal_active_memtable_add_memtable", true /* oneshot */);
-        utils::get_local_injector().enable("table_seal_active_memtable_start_op", true /* oneshot */);
-        utils::get_local_injector().enable("table_seal_active_memtable_try_flush", true /* oneshot */);
+        utils::get_local_injector().enable("table_seal_active_memtable_add_memtable");
```

Comment:
1. it may make it "too easy" for the test to pass if we inject many errors, reducing test selectivity
2. it may pollute things further on (but I see you remembered to disable the injections)

Can we change the condition so it works even if just one memtable fails to flush?